### PR TITLE
feat: DAH-2560 HCBS application question page for The Kelsey

### PR DESCRIPTION
**Pull request by jimlin-sfgov**
## Description

Add a new application question about home and community based services for The Kelsey listing. Text and UI of the question page are not in scope here.

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2560

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] [if the set of changes cannot be small, reviewers have been forewarned](https://google.github.io/eng-practices/review/developer/small-cls.html#cant)
- [x] code meets test coverage thresholds
- [x] code is properly formatted
- [x] code is linted
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] automated tests pass
- [x] if the code changes the UI, it matches the UI design exactly

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions have already been performed at least once
- [x] instructions can be followed by PA testers
- [x] instructions specify if it can only be followed by an engineer

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

### CODE REVIEW

**Comment by jimlin-sfgov** [comment]
Note for reviewers: the set of changes spans many files and is potentially confusing (which is typical of Angular updates). To make the review process more manageable, you can review by commit.

The first commit creates the page and inserts it into the proper place in the application flow. The second commit saves the answer to the question to the application.

File: app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee [hunk 1]
```diff
@@ -1146,6 +1147,10 @@ ShortFormApplicationService = (
       ListingDataService.listing.Custom_Listing_Type
     )
 
+  # TODO: The Kelsey - find a better way to ID before this goes live
+  Service.listingHasHomeAndCommunityBasedServicesUnits  = (listing) ->
```
Comment (tallulahkay) [inline app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee:1151]: What about the adding a type to the custom listing type field? That field is currently only used for educator listings

Comment (jimlin-sfgov) [reply app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee:1151]: That would require additional Salesforce work, and an additional instruction for the person setting up the listing in prod. It is one potential solution, but not in scope of this PR.

**Review by tallulahkay — APPROVED** [review]
This code looks good and works well! Good job!!

**Comment by jimlin-sfgov** [comment]
Flaky Angular e2e tests, no angular changes since the last successfuly Angular e2e test. Bypassing branch protections to merge.

**Pull request merged**